packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,228 @@
+   */
+  defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The Select identifier. Links the htmlFor of the field to the select identifier.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Whether the input is invalid.
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Whether the field displays as Incomplete. Use when no value has been provided. _(usage note: `required` must also be set)_.
+   */
+  isIncomplete: PropTypes.bool,
+  /**
+   * Callback function triggered when the select value changes. function(value)
+   */
+  onChange: PropTypes.func,
+  /**
+   * The array of select options and opt groups.
+   */
+  options: PropTypes.arrayOf(PropTypes.oneOfType([optionPropType, optGroupPropType])),
+  /**
+   * Placeholder data. The presence of this data object is used as the indicator for whether or not a placeholder is used.
+   */
+  placeholder: PropTypes.shape({
+    /**
+     * Whether the placeholder is included as an option to clear selection.
+     */
+    allowClear: PropTypes.bool,
+    /**
+     * Optional override of the default display.
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => options.map(current => {
+  if (current.childOptions) {
+    return (
+      <optgroup key={`${current.display} ${current.childOptions.length}`} label={current.display}>
+        {createOptions(current.childOptions)}
+      </optgroup>
+    );
+  }
+  return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+});
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [currentValue, setCurrentValue] = useState(defaultValue || getFirstValue(options, placeholder));
```

Comment:
What would happen if, after the initial render, the NativeSelect was provided with a new options prop that doesn't include the current, er, `currentValue`? That might be an extreme edge case, but I'm not sure how the native select handles this scenario. 

If it will be a problem, you could validate the currentValue against the options during render and add an effect that resets the state to the new first value.

Reply:
Updated to fix gap.